### PR TITLE
No conv bn folding in ipex to avoid warning

### PR DESCRIPTION
**Pull request by sanderland**
# What does this PR do?

Removes attempting convolution-batchnorm folding from ipex optimization, as it always fails and throws a warning.
See https://github.com/intel/intel-extension-for-pytorch/issues/250

Most models won't even benefit from attempting, but note that even a model like ResNet fails due to checks like:
```
if num_channels != self.num_channels:
```

## Who can review?

- trainer: @sgugger

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Could you check this change @jianan-gu ?

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1280,12 +1280,12 @@ def ipex_optimize_model(self, model, training=False, dtype=torch.float32):
...
         if not training:
             model.eval()
-            model = ipex.optimize(model, dtype=dtype, level="O1")
+            model = ipex.optimize(model, dtype=dtype, level="O1", conv_bn_folding=False)
         else:
             if not model.training:
                 model.train()
             model, self.optimizer = ipex.optimize(
-                model, dtype=dtype, optimizer=self.optimizer, inplace=True, level="O1"
+                model, dtype=dtype, optimizer=self.optimizer, inplace=True, level="O1", conv_bn_folding=False
```
Comment (jianan-gu) [inline src/transformers/trainer.py:1288]: For training, we do not have to add this flag since it is only for inference ([API doc](https://intel.github.io/intel-extension-for-pytorch/latest/tutorials/api_doc.html#ipex.optimize))

**Comment by jianan-gu** [comment]
> 

Thanks for pointing this out.
Actually,  IPEX is doing optimizations as much as it could, and convolution-batchnorm folding does have a chance to bring benefit to vision-related models, e.g, beit/resnet/data2vec_vision, etc. 
Hence if we set it as false, there could be losing some potential optimization chances. 
Back for the warning message, it could be improved on the IPEX side.

**Comment by sanderland** [comment]
> Thanks for pointing this out. Actually, IPEX is doing optimizations as much as it could, and convolution-batchnorm folding does have a chance to bring benefit to vision-related models, e.g, beit/resnet/data2vec_vision, etc. Hence if we set it as false, there could be losing some potential optimization chances. 

The philosophy in the huggingface models seems to be to do a lot of input checks, which is incompatible with the tracing used in ipex. I could not find a single model which doesn't fail.
Discussion on whether or not this tracing could be improved in ipex does not seem to have much traction, see the linked issue.

> Back for the warning message, it could be improved on the IPEX side.

This is certainly true as well.

File: src/transformers/trainer.py [hunk 2]
```diff
@@ -1280,7 +1280,7 @@ def ipex_optimize_model(self, model, training=False, dtype=torch.float32):
 
         if not training:
             model.eval()
-            model = ipex.optimize(model, dtype=dtype, level="O1")
+            model = ipex.optimize(model, dtype=dtype, level="O1", conv_bn_folding=False)
```
Comment (jgong5) [inline src/transformers/trainer.py:1283]: Can you add a comment on how `conv_bn_folding` is disabled explicitly here?

**Review by sgugger — APPROVED** [review]
So let's leave it as `False` for now and revisit once IPEX has better support?

**Comment by jgong5** [comment]
> So let's leave it as `False` for now and revisit once IPEX has better support?

Yes, will record this enhancement as TODO for IPEX.
